tests/mdf/test_mdf.py

```diff
@@ -113,24 +168,29 @@ def test_write_json_file(
     input_dict_str,
     simple_edge_format,
 ):
+    comp_inputs = {composition_name: input_dict_str}
+
     # Get python script from file and execute
-    filename = os.path.join(os.path.dirname(__file__), filename)
-    with open(filename, 'r') as orig_file:
-        exec(orig_file.read())
-        exec(f'{composition_name}.run(inputs={input_dict_str})')
-        orig_results = eval(f'{composition_name}.results')
+    orig_script = read_defined_model_script(filename)
+    orig_results, orig_globals, orig_locals = get_model_results_and_state(
+        orig_script, comp_inputs
+    )
 
     # reset random seed
     pnl.core.globals.utilities.set_global_seed(0)
 
     # Save json_summary of Composition to file and read back in.
     json_filename = filename.replace('.py','.json')
-    exec(f'pnl.write_json_file({composition_name}, json_filename, simple_edge_format=simple_edge_format)')
-    exec(pnl.generate_script_from_json(json_filename))
-    # exec(f'{composition_name}.run(inputs={input_dict_str})')
-    exec(f'pnl.get_compositions()[0].run(inputs={input_dict_str})')
-    final_results = eval(f'{composition_name}.results')
-    assert pnl.safe_equals(orig_results, final_results)
+    exec(
+        f'pnl.write_json_file({composition_name}, "{json_filename}", simple_edge_format={simple_edge_format})',
```

Comment:
Should this use `write_mdf_file`? There's a warning that `write_json_file` is deprecated.
Is this writing a new file to the source tree? would it be better to use tmpdir/tmppath fixture and store the temp file elsewhere? If the writing silently fails the test could still pass if there's a stale .json from the previous run, no?

Reply:
Yes, good point, thanks. Using tempfile module

Reply:
pytest has some useful fixtures for handling tempfiles: https://docs.pytest.org/en/6.2.x/tmpdir.html

Reply:
Ah ok I see what you meant. Replaced with the tmp_path fixture